### DISCUSSION TITLE
Does leftwm auto lock/logoff after a ccertain time or is my program ceashing.

### DISCUSSION THREAD

**Discussion by NicTanghe** [leftwm/leftwm#873] [Q&A]
Does leftwm auto lock/logoff after a certain time or is my program crashing.

Also, how would I disable such behaviour, so I can leave my workstation rendering.

**Comment by VuiMuich** [comment]
No, leftwm has no such feature, you may want to look into power management, like this article on the arch wiki:
https://wiki.archlinux.org/title/Power_management
It is possible, that something causes the session to crash, when waking up from sleep.

This might be also helpful: https://wiki.archlinux.org/title/Display_Power_Management_Signaling

Unfortunatelly you probably will need to do some research, what is actually going wrong and if leftwm is crashed by some event or if it is something else.

**Reply by NicTanghe** [reply]
THX for the pointer.
i will add

/etc/systemd/sleep.conf
```
[Sleep]
AllowSuspend=no
AllowHibernation=no
AllowSuspendThenHibernate=no
AllowHybridSleep=no
```

Btw, a crash that simply closes the program on gnome is logging off my session on leftwm.
probably related to vram.
Is there anny way to produce usable logs for this ?

**Comment by VuiMuich** [comment]
I guess we need even more details. 

What program is the initially crashing one?
Which leftwm version are you using? (You can get this info from `leftwm-check`)
How are you starting leftwm or gnome? Possibly not all environment variables that autostart items gnome uses are being set in your leftwm session.

Raising loglevels is don by setting the environment variable `RUST_LOG` default is `info` which prints `error`, `warn` and `info`, raise it by setting it to `debug` or `trace`. The output depends on your leftwm version: since 0.4.0 you get default logging to systemd logs, so you can read the logs with `journald -ft leftwm-worker` in any older version you need to build manually with the build-flag `--features=journald` to enable the same logging features or even uncomment some code line if you want to log to file.
